File: frontend/types/trip.ts
import { components } from "@/generated/types";


export type CreateTripRequest = components["schemas"]["CreateTripRequest"];
export type TripResponse = components["schemas"]["TripResponse"];


export type Trip = {
    title: string;
    startDate: string;
    endDate: string;
    shortDescription: string;
    description: string;
}

File: backend_old/internal/domain/trip.go
package domain

import "time"

type TripStatus string

const (
	TripStatusPlanned   TripStatus = "planned"
	TripStatusOngoing   TripStatus = "ongoing"
	TripStatusCompleted TripStatus = "completed"
	TripStatusCancelled TripStatus = "cancelled"
)

type Trip struct {
	ResourceMeta
	Title            string
	ShortDescription string
	Description      string
	StartDate        time.Time
	EndDate          time.Time
	Status           TripStatus
}
